### ISSUE TITLE
SearchURL containing file://… fails with "Error: Illegal URL"

### ISSUE THREAD

**Issue by whnr** [tridactyl/tridactyl#1415]
*   Brief description of the problem: It is not possible to open local `file:///…` URLs if they are defined as a `searchurl` in tridactylrc.
 I only noticed this for searchurls. Running `:open file:///…` works fine.
*   Tridactyl version (`:version`): 1.14.9
*   Firefox version: 66.0.1
*   Relevant local URLs: file:///usr/share/doc/python/html/search.html?q=%s
*   Contents of ~/.tridactylrc or ~/.config/tridactyl/tridactylrc (if they exist):

```
set searchurls.py  file:///usr/share/doc/python/html/search.html?q=%s
```
*   Operating system: Linux 5.0.4-arch1-1-ARCH

**Comment by glacambre** [comment]
While `tabopen py keyword` can be made to work here (this would require adding a special case in [tabopen](https://github.com/tridactyl/tridactyl/blob/1f12c3161596c5f3cd12d8df3127ddb3c01d06f9/src/excmds.ts#L1916) to check if the query is using a searchurl for a non-openable url and performing interpolation like in [openInTab](https://github.com/tridactyl/tridactyl/blob/1f12c3161596c5f3cd12d8df3127ddb3c01d06f9/src/lib/webext.ts#L190)), `open py keyword` will never work :/

**Comment by bovine3dom** [comment]
Can't we just redirect it to a new tab like we do with `about:` pages?

**Comment by whnr** [comment]
So the problem is just the .py ending? Then it’s an easy fix. I did not think of that.

**Comment by bovine3dom** [comment]
No, the problem is that it's a file:/// URI. What do you mean by the .py ending? `set searchurls.py ...` -> `open py ...`, not `open searchurls.py ...`.

**Comment by glacambre** [comment]
>Can't we just redirect it to a new tab like we do with about: pages?

Yeah, we could. But then it's not opening in the current tab so it's not really `:open` :)

**Comment by Jab2870** [comment]
I don't know if it's related but I also can't open `data:` urls from the tridactyl prompt.

In my case I'm trying to open something in the form

```
data:text/html;base64,<base64 string>
```

**Comment by bovine3dom** [comment]
We're not allowed to do that: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/create (on phone so can't link to anchor - scroll down to URL).

You should be able to do it if you have `:native` installed.

**Comment by Jab2870** [comment]
Thank you so much for your quick response! That makes perfect sense. I am just getting started with tridactyl.
